Sources/AccountSDKIOSWeb/Client/Client.swift

```diff
@@ -224,21 +225,66 @@ public class Client: CustomStringConvertible {
     func destroySession() {
         sessionStorage.remove(forClientId: configuration.clientId)
     }
+    
+    func switchToSharedKeychain(appIdentifierPrefix: String?, completion: @escaping (Result<Void, Error>) -> Void) {
+        guard let appIdentifierPrefix = appIdentifierPrefix, !appIdentifierPrefix.isEmpty else {
+            completion(.failure(SimplifiedLoginManager.SimplifiedLoginError.appIdentifierPrefixNotProvided))
+            return
+        }
+        
+        let sharedAccessGroup = "\(appIdentifierPrefix).\(Self.sharedKeychainGroup)"
+        let sharedKeychain = KeychainSessionStorage(service: Self.keychainServiceName, accessGroup: sharedAccessGroup)
+        
+        let userSessionArray = sessionStorage.getAll() //get all or get for clientID ?
+        migrateToSharedKeychain(userSessionArray: userSessionArray, sharedKeychain: sharedKeychain) { result in
+            switch (result) {
+            case .success(_):
+                let oldKeychain = self.sessionStorage
+                self.sessionStorage = sharedKeychain
+                oldKeychain.remove(forClientId: self.configuration.clientId)
+                completion(.success())
+            case .failure(let error):
+                SchibstedAccountLogger.instance.error("Cannot switch to shared Keychain with error: \(error.localizedDescription)")
+                completion(.failure(error))
+            }
+        }
+    }
+    
+    private func migrateToSharedKeychain(userSessionArray: [UserSession], sharedKeychain: KeychainSessionStorage, completion: @escaping (Result<Void, Error>) -> Void) {
+        for item in userSessionArray {
+            sharedKeychain.store(item) { result in
+                switch (result) {
+                case .success(_):
+                    break
+                case .failure(let error):
+                    SchibstedAccountLogger.instance.error("\(error.localizedDescription)")
+                    completion(.failure(error))
```

Comment:
Will the outer for loop stop looping when we call this completion? Might be a small 🐛  here